andela/travel_tool_front
# Trip planning initiative front

#### What does this PR do?
This PR Implements the trip planning features.

#### Description of Task to be completed?
- Create travel checklists
- Edit travel checklists
- Delete travel checklists
- Display travel checklists for `Open` requests
- Display travel checklists submission percentage

#### How should this be manually tested?
- A user should be able to view a travel checklist for an `open` request
      - Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
      - Change into the project directory
      - Run the server `make start`
      - Login and make a `new request`
      - Click on the `new request` ellipsis and then click on `travel checklist` on the modal dropdown
      - The travel checklist modal should pop up and show the checklist for the travel request.
- A Travel Admin/Super Admin for a location
    -- should be able to edit a previously set travel checklist and the corresponding output of each 
        checklist item
    -- should be able to delete a checklist item
    -- should be able to create a checklist item

#### Any background context you want to provide?
- You need to clone the [backend](https://github.com/andela/travel_tool_back.git) to test this functionality.

- **Add yourself as travel admin**
   -- Add your email address in the .env file on the backend as `DEFAULT_ADMIN=` and start/restart your server
   -- Grab your token from the browser after login and add the token as authorization on postman
   -- Send a put request to `http://localhost:5000/api/v1/user/admin`

#### What are the relevant pivotal tracker stories?
[#161044514](https://www.pivotaltracker.com/story/show/161044514)
[#161044557](https://www.pivotaltracker.com/story/show/161044557)
[#161044598](https://www.pivotaltracker.com/story/show/161044598)
[#161044635](https://www.pivotaltracker.com/story/show/161044635)
[#161044756](https://www.pivotaltracker.com/story/show/161044756)


#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-10-13 at 6 22 22 pm" src="https://user-images.githubusercontent.com/27569442/46908079-09c57b00-cf15-11e8-97dd-f7b8e7bc1a30.png">

<img width="1433" alt="screen shot 2018-10-13 at 6 23 23 pm" src="https://user-images.githubusercontent.com/27569442/46908096-6163e680-cf15-11e8-85b2-92842b5e174b.png">

#### Questions:
N/A
